File: app/src/main/java/com/teamunemployment/lolanalytics/FrontPage/Tabs/MatchHistoryTab/MatchHistoryAdapter.java
package com.teamunemployment.lolanalytics.FrontPage.Tabs.MatchHistoryTab;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.teamunemployment.lolanalytics.Data.model.MatchIdWrapper;
import com.teamunemployment.lolanalytics.FrontPage.Tabs.MatchHistoryTab.Cards.MatchHistoryCardView;
import com.teamunemployment.lolanalytics.FrontPage.Tabs.PlayerAnalysisTab.Model.StatDefinition;
import com.teamunemployment.lolanalytics.FrontPage.Tabs.StatsComparisonTab.Model.BarChartModel;
import com.teamunemployment.lolanalytics.FrontPage.Tabs.StatsComparisonTab.Model.CardData;
import com.teamunemployment.lolanalytics.FrontPage.Tabs.TabContract;
import com.teamunemployment.lolanalytics.R;

import java.util.ArrayList;

/**
 * @author Josiah Kendall.
 *
 * Adapter for the Match History tab view.
 */
public class MatchHistoryAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> implements TabContract.TabAdapter{

    private ArrayList<MatchIdWrapper> matchIds;

    private MatchHistoryPresenter matchHistoryBasePresenter;
    private BarChartModel barChartModel;

    public MatchHistoryAdapter(MatchHistoryPresenter matchHistoryBasePresenter, BarChartModel barChartModel) {
        this.matchHistoryBasePresenter = matchHistoryBasePresenter;
        this.barChartModel = barChartModel;
    }

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.match_history_card, parent, false);
        MatchHistoryCardView matchHistoryCardView = new MatchHistoryCardView(v,parent.getContext(), matchHistoryBasePresenter, barChartModel);
        return matchHistoryCardView;
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {

        // Our card view instance
        MatchHistoryCardView cardView = (MatchHistoryCardView) holder;

        if (matchIds == null) {
            throw new IllegalStateException("setPlayerAnalysisAdapterData() must be called before binding adapter");
        }

        // Load our data
        cardView.loadData(matchIds.get(position).getMatchId());
    }

    @Override
    public int getItemCount() {
        return matchIds.size();
    }

    public void setData(ArrayList<MatchIdWrapper> matchIds) {
        this.matchIds = matchIds;
    }

    @Override
    public void setMatchHistoryAdapterData(ArrayList<MatchIdWrapper> matchIds) {
        this.matchIds = matchIds;
    }

    @Override
    public void setPlayerAnalysisAdapterData(ArrayList<StatDefinition> statDefinitions) {
        throw new IllegalStateException("This is a matchHistoryAdapter. Please use the appropriate method.");
    }

    @Override
    public void setStatComparisonAdapterData(ArrayList<CardData> cardDatas) {
        throw new IllegalStateException("This is a matchHistoryAdapter. Please use the appropriate method.");
    }
}

File: app/src/main/java/com/teamunemployment/lolanalytics/front_page/Tabs/MatchHistoryTab/MatchHistoryAdapter.kt
package com.teamunemployment.lolanalytics.front_page.Tabs.MatchHistoryTab

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup

import com.teamunemployment.lolanalytics.data.model.MatchIdWrapper
import com.teamunemployment.lolanalytics.front_page.Tabs.MatchHistoryTab.Cards.MatchHistoryCardView
import com.teamunemployment.lolanalytics.front_page.Tabs.MatchHistoryTab.Model.BarChartModel
import com.teamunemployment.lolanalytics.front_page.Tabs.TabContract
import com.teamunemployment.lolanalytics.R
import kotlinx.android.synthetic.main.match_history_card.view.*

import java.util.ArrayList

/**
 * @author Josiah Kendall.
 *
 * Adapter for the Match History tab view.
 */
class MatchHistoryAdapter(private val matchHistoryPresenter: MatchHistoryPresenter) : RecyclerView.Adapter<RecyclerView.ViewHolder>(), TabContract.TabAdapter {

    private lateinit var matchIds: ArrayList<String>

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.match_history_card, parent, false)
        return MatchHistoryCardView(v, parent.context, matchHistoryPresenter)
    }

    override fun onBindViewHolder(holder : RecyclerView.ViewHolder, position: Int) {

        // Our card view instance
        val cardView = holder as MatchHistoryCardView
        setClicklistenersUp(cardView)
        // Load our value
        matchHistoryPresenter.loadCardData(matchIds[position], cardView, "todo region")
    }

    private fun setClicklistenersUp(cardView: MatchHistoryCardView) {
        cardView.itemView.cardDetailsButton.setOnClickListener({matchHistoryPresenter.onDetailsButtonClick()})
    }

    override fun getItemCount() : Int {
        return matchIds.size
    }

    fun setData(matchIds: ArrayList<String>) {
        this.matchIds = matchIds
    }

    override fun setMatchHistoryAdapterData(matchIds: ArrayList<String>) {
        this.matchIds = matchIds
    }
}
